spec/index.bs

```diff
@@ -726,8 +714,49 @@ interface GPUBufferUsage {
 };
 </script>
 
+In order to guarantee immediate, efficient, and portable mapping, buffers
+marked for {{GPUBufferUsage/MAP_WRITE}} are only writable by the CPU,
+and cannot be also marked for {{GPUBufferUsage/COPY_DST}} or {{GPUBufferUsage/STORAGE}}.
+
 ## Buffer Mapping ## {#buffer-mapping}
 
+<script type=idl>
+typedef unsigned long GPUMapFlags;
+interface GPUMap {
+    const GPUMapFlags READ  = 0x0001;
+    const GPUMapFlags WRITE = 0x0002;
+};
+</script>
+
+{{GPUBuffer/map()}} returns null if the device is lost.
+
+Because MAP_WRITE is writable only by the CPU, the contents of the ArrayBuffer
+are unchanged between unmap and map. (Or, if the buffer has not been previously
+mapped, the initial buffer contents.
+
+If the contents of a mapping without {{GPUMap/WRITE}} are changed, on unmap(),
```

Comment:
This is super weird but if I understand correctly the intent is that applications are free to race with the GPU, but it causes a device loss so the result of the race is non-observable?

This spec language doesn't work because an application could start with a buffer full of zeroes, map it, race happily with the GPU for extended periods of time, fill the buffer back with zeros, and unmap.

Reply:
I assume this preserves the behavior we have in our current design, where, if you submit a command buffer which references a mapped resource, the command buffer fails instead. I don't see anything in this proposal about persistent mappings; I only see words about synchronous mappings.